demo/app/app.py

```diff
@@ -316,356 +301,387 @@ def click_map(click_data):
-    :param context: Context data from store.
-    :param history: History data for time series predictors.
-    :param presc: Prescription data from store.
+    :param year: Selected context year.
+    :param lat: Selected context lat.
+    :param lon: Selected context lon.
+    :param sliders: Prescribed slider values.
     :param predictor_name: String name of predictor to use from dropdown.
-    :return: Predicted ELUC and trivia values.
+    :return: Predicted ELUC.
     """
-    context_df = pd.DataFrame.from_records(context)
-    history_df = pd.DataFrame.from_records(history)
-    presc_df = pd.DataFrame.from_records(presc)[LAND_USE_COLS]
+    context = df.loc[year, lat, lon]
+    presc = pd.Series(sliders, index=constants.RECO_COLS)
+
+    # Preprocess presc into diffs
+    presc = presc.combine_first(context[constants.NO_CHANGE_COLS])
+    diff = presc[constants.LAND_USE_COLS] - context[constants.LAND_USE_COLS]
+    diff = diff.rename(constants.COLS_MAP)
+    diff_df = pd.DataFrame([diff])
 
     predictor = None
-    prediction, change = 0, 0
+    prediction = 0
+    if predictor_name == "Random Forest":
+        predictor = random_forest_predictor
+        prediction = predictor.predict(diff_df)
+        return f"{prediction:.4f}"
 
-    if predictor_name == "XGBoost":
-        predictor = XGBoostPredictor()
-        prediction = predictor.run_predictor(context_df[CONTEXT_COLUMNS], presc_df)
+    else:
+        return "0"
 
-    elif predictor_name == "LSTM":
-        predictor = LSTMPredictor()
-        context_merged = pd.concat([history_df, context_df], axis=0)
-        # Sanity check
-        context_merged = context_merged.sort_values(by="time", ascending=True)
-        # TODO: This is gross, clean it up.
-        prediction = predictor.run_predictor(context_merged[CONTEXT_COLUMNS + ALL_DIFF_LAND_USE_COLS], presc_df)
 
-    else:
-        return 0, 0, "Model not connected yet"
-    
+@app.callback(
+    Output("total-em", "children"),
+    Output("tickets", "children"),
+    Output("people", "children"),
+    Input("predict-eluc", "value"),
+    State("year-input", "value"),
+    State("lat-dropdown", "value"),
+    State("lon-dropdown", "value"),
+    prevent_initial_call=True
+)
+def update_trivia(eluc_str, year, lat, lon):
+    """
+    Updates trivia section based on rounded ELUC value.
+    :param eluc_str: ELUC in string form.
+    :param year: Selected context year.
+    :param lat: Selected context lat.
+    :param lon: Selected context lon.
+    :return: Trivia string output.
+    """
+    context = df.loc[year, lat, lon]
+    area = context["cell_area"]
+
     # Calculate total reduction
-    area = context_df["cell_area"].iloc[0]
-    total_reduction = prediction * area
-    
-    return f"{prediction:.4f}", \
-        f"{-1 * total_reduction:,.2f} tonnes CO2", \
-            f"{-1 * total_reduction // CO2_JFK_GVA:,.0f} tickets", \
-                f"{-1 * total_reduction // CO2_PERSON:,.0f} people"
-
-
-def main():
```

Comment:
Removed this main method to reveal app.layout globally which is how gunicorn wants things